### DISCUSSION TITLE
Bluetooth file transfer

### DISCUSSION THREAD

**Discussion by mauroForlimpopoli** [Moddable-OpenSource/moddable#631] [Q&A]
I need to transmit a file of about 200kB. The BLE packet is only some bytes (around 20). May I enlange this dimension until a larger size?
In alternative I should use Classic Bluetooth. Is there some support on this way?
Thank you

**Comment by phoddie** [comment]
Yes, you can request a larger MTU size to increase the number of payload bytes in a transfer.  For details see the `exchangeMTU` function and `onMTUExchanged` callback in our [BLE documentation](https://github.com/Moddable-OpenSource/moddable/blob/public/documentation/network/ble/ble.md#exchangemtumtu).

**Reply by mauroForlimpopoli** [reply]
I change the following code to my main:
	onConnected(device) {
		device.exchangeMTU(250);
		this.stopAdvertising();
	}
	onMTUExchanged(device, mtu) {
		trace(`MTU size is now ${mtu}\n`);
		device.discoverAllPrimaryServices();
	}
now I get this error:
C:\Users\fly3\projects\moddable\examples\piu\k6\main.js (963) # Break: UARTServer.prototype.onConnected: no function!

this is my manifest:
{	
	"build": {
		"PARTITIONS_FILE": "./partitions.csv"
	},
	"include": [
		"$(MODDABLE)/examples/manifest_base.json",
		"$(MODDABLE)/examples/manifest_piu.json",
		"$(MODULES)/input/keyboard/manifest.json",
		"$(MODULES)/pins/analog/manifest.json",
		"$(MODDABLE)/modules/files/file/manifest.json",
		"$(MODULES)/network/wifi/manifest.json",
		"$(MODDABLE)/examples/manifest_net.json",
		"$(MODDABLE)/modules/network/ble/manifest_server.json",
		"$(MODDABLE)/modules/network/ble/manifest_client.json",
		"$(MODULES)/pins/i2c/manifest.json",
		"$(MODULES)/network/mdns/manifest.json",
		"$(MODDABLE)/contributed/serial/manifest.json"
	],
	"creation":{
		"stack":500,
		 "static": 98304,
		 "keys": {
			"available":100
		},
		"chunk": {
			"initial":42530,
			"incremental":1024
		}
	},
	"modules": {
		"*": [
			"./main",
			"./assets",
			"$(MODULES)/network/http/*",
			"./restart",
			"./deepSleep",
			"./funcSleep",
			"$(MODULES)/network/sntp/*",
			"$(MODULES)/files/zip/*"
		],
		"pins/pwm": "$(MODULES)/pins/pwm/pwm",
		"esp32/ota": "$(MODDABLE)/build/devices/esp32/modules/ota/*"
	},
	"resources":{
		"*-alpha": [
			"$(MODDABLE)/examples/assets/fonts/OpenSans-Regular-20",
			"$(MODDABLE)/examples/assets/fonts/OpenSans-Semibold-18",
			"$(MODDABLE)/examples/assets/fonts/OpenSans-Light-42",
			"$(MODDABLE)/examples/assets/fonts/OpenSans-Semibold-28",
			"$(MODDABLE)/examples/assets/fonts/OpenSans-Regular-72",
			"$(MODDABLE)/examples/assets/fonts/arc64"
		],
		"*-image(50)": [
			"$(MODDABLE)/examples/assets/images/back-icon",
			"$(MODDABLE)/examples/assets/images/Bluetooth-icon",
			"$(MODDABLE)/examples/assets/images/Wifi-icon",
			"$(MODDABLE)/examples/assets/images/unlock-icon",
			"$(MODDABLE)/examples/assets/images/lock-icon"			
		],
		"*-mask": [
			"$(MODDABLE)/examples/assets/images/loading-icon",
			"$(MODDABLE)/examples/assets/images/thermometer-mask",
			"$(MODDABLE)/examples/assets/images/battery-mask"		
		]
	},
	"platforms": {
		"esp": {
			"config": {
				"rotation": 90
			}
		},
		"esp32": {
			"config": {
				"rotation": 90,
				"brightness": 0,
				"file": {
					"root": "/mod/"
				}
			},
			"modules": {
				"*": [	
				"$(MODULES)/pins/pwm/esp32/*",
				"$(MODULES)/files/flash/*",
				"$(MODULES)/files/flash/esp32/*"
				]
			},
			"defines": {
				"serial": {
					"interface": { "UART": 2 },
					"baud": 9600,
					"config": { "dataBits": 8, "parity": "#N", "stopBits": 1 },
					"tx_pin": 17,
					"rx_pin": 16
				}
			}
		}
	},
	"preload": [
		"http",
		"sntp",
		"pins/i2c",
		"file",
		"pins/analog",
		"timer",
		"esp32/ota",
		"ble",
		"zip"
	],
	 "data": {
        "*": [
            "./mydata",
			"./myWeb"			
        ]
    },"ble":{
		"*": [
			"./bleservices/*"
		]
	}
}
and this is the usart.json of the ble:
{
	"service": {
		"uuid": "6E400001-B5A3-F393-E0A9-E50E24DCCA9E",
		"characteristics": {
			"rx": {
				"uuid": "6E400002-B5A3-F393-E0A9-E50E24DCCA9E",
				"type": "String",
				"maxBytes": 512,
				"permissions": "write",
				"properties": "write"
			},
			"tx": {
				"uuid": "6E400003-B5A3-F393-E0A9-E50E24DCCA9E",
				"type": "String",
				"maxBytes": 512,
				"permissions": "read",
				"properties": "notify"
			}
		}
	}
}
Do I need to include other modules ?

**Reply by phoddie** [reply]
It is a little challenging to diagnose a report from several source code files merged together without formatting. Would you provide a link to a repository or Gist with the full project? Thank you.

**Reply by mauroForlimpopoli** [reply]
This is the full project. I'm sorry that under development and it's very confused.
It create a BLE server (Uart), when you try to connnet to it, there is the problem:
C:\Users\fly3\projects\moddable\examples\piu\k6\main.js (963) # Break: UARTServer.prototype.onConnected: no function!
[prog.zip](https://github.com/Moddable-OpenSource/moddable/files/6321256/prog.zip)
